WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerSearchView.java

```diff
@@ -64,9 +63,7 @@ public boolean onMenuItemActionCollapse(MenuItem item) {
     public void enableSearchMode() {
         mMenuEdit.setVisible(false);
         mSitePickerActivity.setIsInSearchModeAndNullifyAdapter(true);
-        SitePickerSearchAdapter adapter = (SitePickerSearchAdapter) mSitePickerActivity.getAdapter();
-        mSitePickerActivity.getRecycleView().swapAdapter(adapter, true);
-        adapter.loadSites();
+        mSitePickerActivity.getRecycleView().swapAdapter(mSitePickerActivity.getAdapter(), true);
```

Comment:
You could continue to do it using `onSaveInstanceState()` as you do know, but have getters/setters for those values in the adapter itself. Then in `onCreate()` you'd have something along these lines:

```
boolean isInSearchMode = false;
String lastSearch = null;

if (savedInstanceState != null) {
    mCurrentLocalId = savedInstanceState.getInt(KEY_LOCAL_ID);
    isInSearchMode = savedInstanceState.getBoolean(KEY_IS_IN_SEARCH_MODE);
    lastSearch = savedInstanceState.getString(KEY_LAST_SEARCH);
} else if (getIntent() != null) {
    mCurrentLocalId = getIntent().getIntExtra(KEY_LOCAL_ID, 0);
}

setupFab();

mRecycleView = (RecyclerView) findViewById(R.id.recycler_view);
mRecycleView.setLayoutManager(new LinearLayoutManager(this));
mRecycleView.setScrollBarStyle(View.SCROLLBARS_OUTSIDE_OVERLAY);
mRecycleView.setItemAnimator(null);
mRecycleView.setAdapter(getAdapter());

if (isInSearchMode) {
    getAdapter().searchSites(lastSearch);
}
```

If you think that would be more error-prone, we can stick with it the way it is now. I just worry that down the road the duplication could be an issue.